### PR TITLE
Bump starlette to 0.20.4

### PR DESCRIPTION
**Pull request by ShayBox**
Security vulnerability was fixed in 0.20.3

### CODE REVIEW

**Comment by ShayBox** [comment]
starlette@0.20.0 dropped support for Python 3.6 https://github.com/encode/starlette/pull/1616

File: fastapi/security/api_key.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def __init__(
         self.auto_error = auto_error
 
     async def __call__(self, request: Request) -> Optional[str]:
-        api_key: str = request.query_params.get(self.model.name)
+        api_key = request.query_params.get(self.model.name)
```
Comment (JarroVGIT) [inline fastapi/security/api_key.py:30]: Why would you remove the typing?

**Review by JarroVGIT — COMMENTED** [review]
Why would you remove the typing of api_key?

**Comment by ShayBox** [comment]
The test failed and that's what the previous pr for 0.20.2 did by Kludex

File: fastapi/security/api_key.py (hunk 1 above)
Comment (ShayBox) [reply fastapi/security/api_key.py:30]: The test failed and that's what the previous pr for 0.20.2 did by @Kludex

**Comment by bonastreyair** [comment]
Last starlette version that supported python 3.6 is 1.19.1 https://github.com/encode/starlette/blob/e3de71f1433568a3f7ac3dc69ef86cf4f72f5651/README.md?plain=1#L44

so verison 0.20.4 will never pass the tests since it requires Python >=3.7
https://github.com/encode/starlette/blob/e3de71f1433568a3f7ac3dc69ef86cf4f72f5651/setup.py#L27

**Comment by ShayBox** [comment]
Yeah this version change would require following and dropping Python 3.6 support

**Comment by mdczaplicki** [comment]
Any chances to push this PR?

**Comment by ShayBox** [comment]
Python 3.6 reached EOL and stopped getting security patches 7 months ago, so it's probably for the best to drop support.

**Comment by Ancieg** [comment]
Push this PR, please.

**Comment by ShayBox** [comment]
Nobody closed this pull, version has been bumped and Python 3.6 has been dropped.

**Pull request closed without merging**